packages/dashboard/src/layout/usePanelRegistration.ts

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import {
+  DashboardPluginComponentProps,
+  PanelComponentType,
+  PanelDehydrateFunction,
+  PanelHydrateFunction,
+  PanelProps,
+} from '../DashboardPlugin';
+
+/**
+ * Registers a given panel component. Also runs a `useEffect` that will
+ * automatically de-register then panel on unmount.
+ * @param registerComponent
+ * @param ComponentType
+ * @param hydrate
+ * @param dehydrate
+ */
+export default function usePanelRegistration<
+  P extends PanelProps,
+  C extends React.ComponentType<P>
+>(
+  registerComponent: DashboardPluginComponentProps['registerComponent'],
+  ComponentType: PanelComponentType<P, C> | React.NamedExoticComponent<P>,
+  hydrate?: PanelHydrateFunction<P>,
+  dehydrate?: PanelDehydrateFunction
+) {
+  // Look for static `COMPONENT` or `displayName` attribute
+  const name =
+    'COMPONENT' in ComponentType
+      ? ComponentType.COMPONENT
+      : ComponentType.displayName;
```

Comment:
I think you can just use null coalescing to handle this:
```suggestion
  const name = ComponentType.COMPONENT ?? ComponentType.displayName;
```

Reply:
I actually did that originally, but the type system complains when you access `ComponentType.COMPONENT` without verifying it's there

Reply:
nvm, fixed it in the types